<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <widget class="QTreeView" name="treeView">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>40</y>
      <width>481</width>
      <height>341</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="addButton">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>50</y>
      <width>88</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Add</string>
    </property>
   </widget>
   <widget class="QPushButton" name="removeButton">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>100</y>
      <width>88</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Remove</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="spinBox">
    <property name="geometry">
     <rect>
      <x>700</x>
      <y>80</y>
      <width>44</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="findButton">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>140</y>
      <width>88</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Find</string>
    </property>
   </widget>
   <widget class="QTextBrowser" name="answerBrowser">
    <property name="geometry">
     <rect>
      <x>40</x>
      <y>400</y>
      <width>201</width>
      <height>91</height>
     </rect>
    </property>
   </widget>
   <widget class="QPushButton" name="traverseButton">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>190</y>
      <width>88</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Traverse</string>
    </property>
   </widget>
   <widget class="QPushButton" name="insertSubtreeButton">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>240</y>
      <width>101</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Insert Subtree</string>
    </property>
   </widget>
   <widget class="QPushButton" name="removeSubtreeButton">
    <property name="geometry">
     <rect>
      <x>570</x>
      <y>290</y>
      <width>121</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Remove Subtree</string>
    </property>
   </widget>
   <widget class="QPushButton" name="commonParentButton">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>410</y>
      <width>221</width>
      <height>25</height>
     </rect>
    </property>
    <property name="text">
     <string>Find the closest common parent</string>
    </property>
   </widget>
   <widget class="QSpinBox" name="value1SpinBox">
    <property name="geometry">
     <rect>
      <x>280</x>
      <y>450</y>
      <width>44</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
   <widget class="QSpinBox" name="value2SpinBox">
    <property name="geometry">
     <rect>
      <x>360</x>
      <y>450</y>
      <width>44</width>
      <height>26</height>
     </rect>
    </property>
   </widget>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>22</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <resources/>
 <connections/>
</ui>
